### PR TITLE
refactor: [M3-6343] - MUI v5 Migration - Grid v2 for Features

### PR DESCRIPTION
**Pull request by jaalah-akamai**
## Description 📝
Migrating to MUI Grid v2. This covers everything in the `src/features` directory (excluding Longview which is forthcoming) and any additional files that needed to be cleaned up.

> **Note**: In several places, you _MAY_ notice slight padding variations due to overly excessive `Grid` usage. In places like this, I made my best judgment and consulted with UX/Design. You'll notice in several places the use of `padding: 0` on `Grid` elements. This was to match the existing design as much as possible until we get a chance to revisit these areas. Feel free to highlight them, and I'll comment!

## Major Changes 🔄 
- `MainContent.tsx` - You'll notice _slightly_ more padding for the main content container. This is because the page is actually respecting the original padding of the container now.
- Changed `Grid` import from `src/components/Grid` to `@mui/material/Unstable_Grid2`. 
   - The `item` attribute is no longer needed for `Grid`.
   - Most `Grid` elements with the `container` attribute needed a `spacing` attribute as part of the change
- `updateFor` has been removed off the Grid component. There's no need to micromanage how/when a component should re-render like this anymore.

### Refactored
- `LinodeEntityDetail.tsx` - Very large file, did some major refactoring here to get things looking like they did before. Check all breakpoints. You'll notice that `DNS Resolvers` has been updated to use CSS Grid and its layout on smaller breakpoints is slightly different. UX approves 👍 
- `DNSResolvers.tsx` - Simplified code and removed `makeStyles`
- `NodeBalancerDetail.tsx` - Using `LandingHeader` and removed `makeStyles`
- `RenderNotification.tsx` - No need for Grid, this was an easy refactor
- `useFormattedNotifications.tsx` - Due to changes in `RenderNotification` this needed a slight refactor
- `LinodeAdvancedConfigurationsPanel.tsx` - Didn't seem like the styles here were being used
- `AlertSection.tsx` - Instead of forcing changes to work, I saw refactoring was more simple.

### Replaced Grid w/ Box component
There was no need for a `Grid` components in these files, so I swapped them with `Box` or removed the `Grid` completely.
- `ExpandableTicketPanel.tsx` - Removed lots of extra `Grid` components in this file
- `AutoEnroll.tsx`
- `DatabaseSummaryConnectionDetails.tsx`
- `FirewallRuleTable.tsx`
- `SupportSearchLanding.tsx`
- `ImageSelect.tsx`
- `KubeConfigDisplay.tsx`
- `NodeBalancerConfigurations.tsx`
- `TablesPanel.tsx` - This is under NodeBalancers
- `VolumeTableRow.tsx`

### Deleted
- `features/Domains/DomainsTableWrapper.tsx` - This didn't seem to be used

## How to test
1. These changes touch the core features we offer, so a page by page scan is required
2. Go to https://cloud.linode.com/ and compare side by side with local
3. There are slight changes as noted above and you may notice slight padding changes. Anything agregious, just flag in this PR and I'll take a look at.
4. Start with larger breakpoints and work your way down.

<details>
<summary>Where to view changes? Open me! 👇 </summary>
<br>

| Component    | Where to view or Desc? | Screenshot |
| ----------- | ----------- | ----------- |
| AutoEnroll <br /> http://localhost:3000/account/settings | Click on "enable now" link to open drawer. If you're a "managed" user, you may not be able to see this. | |
| DatabaseSummaryConnectionDetails <br /> http://localhost:3000/databases/postgresql/{ID}/summary | | Screen Shot 2023-04-12 at 10 18 49 AM|
| Firewalls Rules <br /> http://localhost:3000/firewalls/{ID} | | Screen Shot 2023-04-12 at 10 21 19 AM |
| ImageSelect <br /> http://localhost:3000/stackscripts/create | | Screen Shot 2023-04-12 at 10 28 02 AM|
| KubeConfigDisplay <br /> http://localhost:3000/kubernetes/clusters/{ID}/summary | | Screen Shot 2023-04-12 at 10 30 27 AM|
| NodeBalancerConfigurations  <br/> http://localhost:3000/nodebalancers/{ID}/configurations | Configurations tab, lots of refactoring here | Screen Shot 2023-04-12 at 10 33 39 AM |
| Linode Summary View <br /> http://localhost:3000/linodes?view=grid | Tags were using a 1400 breakpoint which is not a supported breakpoint. These will now change layout at < 1280 | Screen Shot 2023-04-12 at 9 20 59 AM|
| Linode Detail <br /> http://localhost:3000/linodes/{ID}/networking | DNS Resolvers has been refactored | Screen Shot 2023-04-12 at 9 35 58 AM|
| Notifications <br /> http://localhost:3000/profile/settings | Use mock service worker > open events dropdown | Screen Shot 2023-04-12 at 9 57 00 AM |
| ExpandableTicketPanel <br /> http://localhost:3000/support/tickets | Use MSW | Screen Shot 2023-04-12 at 10 02 05 AM |
| Entity Transfer Create <br /> http://localhost:3000/account/service-transfers/create <br /> http://localhost:3000/account/service-transfers | | |
|Firewalls Linodes <br/> http://localhost:3000/firewalls/{ID}/linodes|| Screen Shot 2023-04-12 at 10 09 24 AM|
| Search Results <br /> http://localhost:3000/support/search/?query=test | | Screen Shot 2023-04-12 at 10 11 00 AM|
| User Menu| Top navigation user menu |  Screen Shot 2023-04-12 at 11 29 52 AM|
</details>

## QA Checklist
Tag your name next to the ones you've QA'd
- [x] Linodes
- [x] Volumes
- [x] NodeBalancers
- [x] Firewalls
- [x] Stackscripts
- [x] Images
- [x] Domains
- [x] Databases
- [x] Kubernetes
- [x] Object Storage
- [x] Marketplace - Shouldn't be any changes here
- [x] Account
- [x] Help & Support

### CODE REVIEW

File: packages/manager/src/features/linodes/LinodesDetail/LinodeAdvanced/LinodeAdvancedConfigurationsPanel.tsx [hunk 1]
```diff
@@ -1,45 +1,20 @@
...
-    marginTop: theme.spacing(),
-  },
-  sidebar: {
-    marginTop: `-${theme.spacing(2)}`,
-    [theme.breakpoints.up('md')]: {
-      marginTop: theme.spacing(5),
-    },
-  },
-}));
-
```
[the remark this answers is not in the document]
Comment (jaalah-akamai) [reply packages/manager/src/features/linodes/LinodesDetail/LinodeAdvanced/LinodeAdvancedConfigurationsPanel.tsx:34]: I could've gone further with this file, but I left it intact for now

File: packages/manager/src/features/linodes/LinodesDetail/LinodeAdvanced/LinodeConfigs.tsx [hunk 2]
```diff
@@ -171,17 +157,19 @@ class LinodeConfigs extends React.Component<CombinedProps, State> {
...
         <Grid
           container
           alignItems="flex-end"
           justifyContent="flex-end"
-          className={classes.root}
+          sx={{
+            padding: '0 0 8px 0',
+          }}
         >
-          <Grid item className={classes.addNewWrapper}>
```
Comment (jaalah-akamai) [inline packages/manager/src/features/linodes/LinodesDetail/LinodeAdvanced/LinodeConfigs.tsx:184]: No need for all the spacing adjustment from `addNewWrapper` anymore.

**Review by bnussman-akamai — COMMENTED** [review]
Things generally look good so far, can we decrease the spacing here across all entities, it has gotten larger in this PR

Screenshot 2023-04-13 at 10 26 12 AM

**Comment by jaalah-akamai** [comment]
> Things generally look good so far, can we decrease the spacing here across all entities, it has gotten larger in this PR
> 
> Screenshot 2023-04-13 at 10 26 12 AM

Yea I was going back and forth with that, but yea let's do it!

**Review by dwiley-akamai — COMMENTED** [review]
I still have a few more places in the app to check and a lot of code to sift through, but it's looking good so far.

One small thing I noticed is that the graphics on the Referrals page are closer together than they were previously, do we want to keep that?

**Review by hana-linode — CHANGES_REQUESTED** [review]
The TagsPanel is no longer flush to the right side

image

Mobile spacing for the firewalls rule table seems a bit tight now
This branch
image

Prod
image

Mobile spacing for Kube display is a bit off/tight
image

"Port Configuration" display disappeared
This branch
image

Prod
image

The extra white padding around the accordion header looks less clean imo
This branch
image

Prod
image

**Comment by bnussman-akamai** [comment]
@hana-linode The removal of `Port Configuration` was done by me in https://github.com/linode/manager/pull/8964, it was intentional

**Comment by jaalah-akamai** [comment]
Thanks @hana-linode let me know if things are looking better now. I actually improved the alignment of the inbound rules dropdown to always match the action column since I saw that was the intent cc: @mjac0bs 


https://user-images.githubusercontent.com/125309814/231877452-286e8fe0-23fb-47b8-8110-53ddaddb4c02.mp4

Screen Shot 2023-04-13 at 4 36 33 PM

Screen Shot 2023-04-13 at 4 38 34 PM

**Comment by jaalah-akamai** [comment]
> 3c6f9ab

Hmm fix incoming!

**Comment by mjac0bs** [comment]
> I actually improved the alignment of the inbound rules dropdown to always match the action column since I saw that was the intent

🥇 @jaalah-akamai Thank you for doing that. I remember it being just slightly off in the accessibility fix and it haunted me a little. Giving the Firewalls page and the rest of this PR a closer look now!

**Comment by hana-linode** [comment]
Ah I probably should have cropped the image more, but I meant that the mobile spacing between table columns looks a bit tight now

This branch
image

Prod
image

**Comment by jaalah-akamai** [comment]
@hana-linode gotcha! should be good now, increased spacing slightly for the first item from `30%` to `32%` because we couldn't use margins to create space like we did in prod. I didn't want any words breaking to new lines as things got smaller.

**Review by mjac0bs — COMMENTED** [review]
I'm still working my way through pages at each breakpoint, but here are a few spacing differences so far:

Firewalls -- This looks better now! I'm happy with the changes here.
~~As @hana-linode pointed out, spacing between columns is tight as we move to smaller screens. This is especially noticeable in columns if a user has a long rule name or list of many IPs.~~ ✅ 
| This branch  | Prod  |  
|---|---|
|  Screenshot 2023-04-13 at 2 00 18 PM | Screenshot 2023-04-13 at 2 00 04 PM  |

Linode Landing --  The collapsed tags "..." and the "Add a Tag" button lack some spacing between so they meld into one.
| This branch  | Prod  |  
|---|---|
| Screenshot 2023-04-13 at 2 11 05 PM  | Screenshot 2023-04-13 at 2 11 11 PM  |

StackScripts
- Thank you for fixing Image Select > Target Images! I recently made a ticket for this because I noticed our wandering TooltipIcons. 

Linodes Details > Storage > Add a Disk - in the same ticket I made for the above, I noticed another spot with the misaligned icon. On this branch, it's still looking like it needs some adjustment on the Image field.
| This branch  | Prod  |  
|---|---|
| Screenshot 2023-04-13 at 2 50 53 PM |  Screenshot 2023-04-13 at 2 51 21 PM |

Images -- Looks good to me, other than: just noticed this minor styling bug while testing (also in prod) that might be out of scope of this PR, but it would be cool if we could get our TooltipIcon on the Region field in Upload Image to stop having this weird  spacing at the `599` breakpoint. 

https://user-images.githubusercontent.com/114685994/231888206-08ef9bbd-ff5c-466c-a56f-e5716687a7a4.mov

**Review by dwiley-akamai — APPROVED** [review]
LGTM 🚀

**Review by mjac0bs — COMMENTED** [review]
**Help and Support:**

At a width of `599px` on down, there is some misalignment happening that I think we could fix here (also currently in prod):
image

**Profile > Login & Authentication:** (https://localhost:3000/profile/auth)
TPA provider options are having some issues now at smaller breakpoints.

Prod:

https://user-images.githubusercontent.com/114685994/232110345-86263e3f-9549-4db8-aeee-2a05586ea3f5.mov

This branch:

https://user-images.githubusercontent.com/114685994/232110321-2597db55-616e-4b1e-8bf7-12067719d4af.mov

**Linode Create:**
|  This branch |  Prod | 
|---|---|
| Screenshot 2023-04-14 at 10 35 04 AM | Screenshot 2023-04-14 at 10 34 49 AM

Kubernetes:
I know we recently did some work on this page... the margin on the right seems to small to me and feels better in prod, but I didn't review the other ticket when this was refactored.

|  This branch |  Prod | 
|---|---|
| Screenshot 2023-04-14 at 10 29 30 AM
 | Screenshot 2023-04-14 at 10 29 37 AM |

**Review by bnussman-akamai — APPROVED** [review]
I think this is in a good enough state to proceed! I think any outstanding issues could be resolved in this PR or after this is merged

**Review by hana-linode — APPROVED** [review]
Other than Mariah's comments, this is looking good to me 🚀

**Comment by jaalah-akamai** [comment]
Here are the updates - thanks everybody!
|||
| ----------- | ----------- |
| <video src="https://user-images.githubusercontent.com/125309814/232144970-f22ca018-3b81-4b4a-b2fe-60703ba36d5f.mp4" />| <video src="https://user-images.githubusercontent.com/125309814/232145051-a719f115-13e0-4280-8655-6010833dbe8c.mp4" />|
| <video src="https://user-images.githubusercontent.com/125309814/232145107-5dcbb717-dcb0-40dd-8c22-e80296ae11bd.mp4" />| <video src="https://user-images.githubusercontent.com/125309814/232145140-5cabc548-a9c8-4c63-b714-8c04508e6bf1.mp4" />|

**Review by mjac0bs — APPROVED** [review]
Thanks for the fixes! 🚀 Can we add this to the Tech Stories section in the changelog?

**Pull request merged**